domenetjenester/datavarehus/src/main/java/no/nav/foreldrepenger/datavarehus/v2/StønadsstatistikkTjeneste.java

```diff
@@ -1,76 +1,365 @@
+        var familiehendelse = familieHendelseTjeneste.hentAggregat(behandlingId).getGjeldendeVersjon();
+
+        var lovVersjon = utledLovVersjon(stp);
         var fagsak = behandling.getFagsak();
-        return new StønadsstatistikkVedtak(map(fagsak.getSaksnummer()), map(fagsak.getYtelseType()), behandling.getUuid(), map(fagsak.getAktørId()),
-            utledRettigheter(behandling));
+        var saksnummer = mapSaksnummer(fagsak.getSaksnummer());
+        var ytelseType = mapYtelseType(fagsak.getYtelseType());
+        var søker = mapAktørId(fagsak.getAktørId());
+        var familieHendelse = mapFamilieHendelse(behandlingReferanse, familiehendelse);
+
+
+        var builder = new Builder().medLovVersjon(lovVersjon)
+            .medSaksnummer(saksnummer)
+            .medSøker(søker)
+            .medYtelseType(ytelseType)
+            .medBehandlingUuid(behandlingReferanse.behandlingUuid())
+            .medForrigeBehandlingUuid(forrigeBehandlingUuid.orElse(null))
+            .medSkjæringstidspunkt(stp.getSkjæringstidspunktHvisUtledet().orElse(null))
+            .medVedtakstidspunkt(vedtak.getVedtakstidspunkt())
+            .medVedtaksresultat(mapVedtaksresultat(vedtak))
+            .medVilkårIkkeOppfylt(utledVilkårIkkeOppfylt(vedtak, behandling))
+            .medUtlandsTilsnitt(utledUtlandsTilsnitt(utlandMarkering))
+            .medAnnenForelder(utledAnnenForelder(behandling, familiehendelse))
+            .medFamilieHendelse(familieHendelse)
+            .medUtbetalingsreferanse(String.valueOf(behandlingReferanse.behandlingId()));
+
+        if (FagsakYtelseType.FORELDREPENGER.equals(fagsak.getYtelseType())) {
+            var rettigheter = utledRettigheter(behandling);
+            var foreldrepengerUttaksperioder = mapForeldrepengerUttaksperioder(behandling, rettigheter.rettighetType());
+            builder.medUttakssperioder(foreldrepengerUttaksperioder).medForeldrepengerRettigheter(rettigheter);
+        }
+        if (FagsakYtelseType.ENGANGSTØNAD.equals(fagsak.getYtelseType())) {
+            builder.medEngangsstønadInnvilget(utledTilkjentEngangsstønad(behandlingId));
+        } else {
+            var utbetalingssperioder = mapUtbetalingssperioder(behandling, ytelseType, familieHendelse.hendelseType());
+            builder.medUtbetalingssperioder(utbetalingssperioder).medBeregning(utledBeregning(behandling));
+        }
+        return builder.build();
+    }
+
+    private Beregning utledBeregning(Behandling behandling) {
+        if (FagsakYtelseType.ENGANGSTØNAD.equals(behandling.getFagsakYtelseType())) {
+            return null;
+        }
+        var beregningsgrunnlag = beregningsgrunnlagTjeneste.hentBeregningsgrunnlagEntitetAggregatForBehandling(behandling.getId());
+        var skjæringstidspunkt = beregningsgrunnlag.getSkjæringstidspunkt();
+
+        var bruttoÅrsinntekt = beregningsgrunnlag.getBeregningsgrunnlagPerioder()
```

Comment:
I enkelte saker kan brutto variere i senere perioder (kortvarig arb.for med avvik og endring i naturalytelse), men ikke sikkert det er interessant her? Dette vil være rett i det store flertallet av saker

Reply:
Jeg tror det man er interessert i er oversikt over "brutto inntekt" og da er STP like greit som noe annet.
Spørsmålet vårt var mer på framgangsmåten vår - plukke STP/B og så finne en BG-periode som overlapper.